source/includes/fundamentals/code-snippets/JMXMonitoring.java

```diff
@@ -0,0 +1,102 @@
+ *
+ *   - Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *   - Neither the name of Oracle or the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.mycompany.app;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.*;
+import com.mongodb.management.JMXConnectionPoolListener;
+
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.rmi.registry.LocateRegistry;
+
+/*
+  Class to demonstrate JMX server for connection pool events in MongoDB Java Driver.
+  Based heavily off of this example from Oracle:
+  https://docs.oracle.com/javase/8/docs/technotes/guides/jmx/examples/Basic/Server.java
+
+  This is the reason the above Oracle
+  copyright notice has been included
+ */
+public class JMXMonitoring {
+
+    private static final String COLLECTION = "compound-test";
+    private static final String DATABASE = "test";
+    private static final ConnectionString URI = new ConnectionString(System.getenv("DRIVER_URL"));
+
+    public static void main(String[] args) throws InterruptedException {
+        // start jmx-example
+        JMXConnectionPoolListener connectionPoolListener = new JMXConnectionPoolListener();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToConnectionPoolSettings(builder -> builder.addConnectionPoolListener(connectionPoolListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        int port = 9999;
+        javax.management.MBeanServer server = ManagementFactory.getPlatformMBeanServer();
```

Comment:
I don't think you need any of this funky code.  The JVM does this for you.  

Just put a `Thread.sleep(Long.MAX_VALUE)` after creating MongoClient, so the program doesn't exit, and then run jconsole.  You should still be able to connect and see all the MBeans.

Reply:
Thanks for the suggestion. Fix: https://github.com/mongodb/docs-java/pull/88/commits/20e629ece2705d227875b2895c8d40de52800e7d

This next paragraph is purely for my own edification (please feel free to ignore):

It seems like the platform Mbean server is automatically exposed by the JVM for local connections when the ``com.sun.management.jmxremote`` property is set. Does the driver set this property? I looked into this for a little bit and couldn't find an answer.